src/export/csv.rs

```diff
@@ -72,6 +73,14 @@ pub fn export_csv(exporter: &mut Exporter) -> Result<()> {
         let native = ctx.map(|&(_, ref n)| n).flatten();
 
         if let Some(curr) = foreign.curr {
+            match end_at {
+                Some(seconds) if curr.period.begin() > seconds => {
+                    println!("Ending at {:?} seconds of video.", seconds);
+                    break;
+                }
+                _ => {}
+            }
+
```

Comment:
Is there a clean way in Rust to extract this 3x duplicated code block? The trick is that it uses `break` it stop iterating early, which a side-effect-y helper function usually can't do. But if I could get rid of the duplication, that would make me happier and the code easier to maintain. :)